test/python/transpiler/test_unitary_synthesis.py

```diff
@@ -843,19 +844,39 @@ def test_iswap_no_cx_synthesis_succeeds(self):
         result_qc = dag_to_circuit(result_dag)
         self.assertTrue(np.allclose(Operator(result_qc.to_gate()).to_matrix(), cxmat))
 
-    def test_parameterized_basis_gate_in_target(self):
-        """Test synthesis with parameterized RXX gate."""
+    @data(True, False)
+    def test_parameterized_basis_gate_in_target(self, is_random):
+        """Test synthesis with parameterized RZZ gate."""
         theta = Parameter("θ")
         lam = Parameter("λ")
+        phi = Parameter("ϕ")
         target = Target(num_qubits=2)
         target.add_instruction(RZGate(lam))
-        target.add_instruction(RXGate(theta))
-        target.add_instruction(RXXGate(theta))
+        target.add_instruction(RXGate(phi))
+        target.add_instruction(RZZGate(theta))
         qc = QuantumCircuit(2)
+        if is_random:
+            qc.unitary(random_unitary(4, seed=1234), [0, 1])
         qc.cp(np.pi / 2, 0, 1)
         qc_transpiled = transpile(qc, target=target, optimization_level=3, seed_transpiler=42)
         opcount = qc_transpiled.count_ops()
-        self.assertTrue(set(opcount).issubset({"rz", "rx", "rxx"}))
+        self.assertTrue(set(opcount).issubset({"rz", "rx", "rzz"}))
+        self.assertTrue(np.allclose(Operator(qc_transpiled), Operator(qc)))
+
+    @data(
+        ["rx", "ry", "rxx"],
+        ["rx", "rz", "rzz"],
+    )
+    def test_parameterized_backend(self, basis_gates):
+        """Test synthesis with parameterized backend."""
+        backend = GenericBackendV2(3, basis_gates=basis_gates)
```

Comment:
I don't think this will have a big effect on this particular test, but as a good practice, I recommend always setting the seed of the generic backend to make sure the errors and durations are always the same (this affects especially the routing tests)

```suggestion
        backend = GenericBackendV2(3, basis_gates=basis_gates, seed=0)
```

Reply:
done in [6a82649](https://github.com/Qiskit/qiskit/pull/13568/commits/6a82649709a9ee23f7eef241c055f0f483f00b79)